src/Rings/FreeAssociativeAlgebraIdeal.jl

```diff
@@ -44,7 +44,7 @@ Return the two-sided ideal of $R$ generated by $g$.
 """
 function ideal(R::FreeAssociativeAlgebra, g::Vector{<:FreeAssociativeAlgebraElem})
   @req all(x -> parent(x) == R, g) "parent mismatch"
-  return FreeAssociativeAlgebraIdeal(IdealGens(R, g))
+  return FreeAssociativeAlgebraIdeal(IdealGens(R, collect(elem_type(R),g)))
 end
```

Comment:
The (admittedly small) problem with your change is that it always allocates a new vector, even if it is not necessary. I suggest something like this instead (untested):

```suggestion
function ideal(R::FreeAssociativeAlgebra, g::Vector{U}) where {U <: FreeAssociativeAlgebraElem}
  @req all(x -> parent(x) == R, g) "parent mismatch"
  h = (U === elem_type(R)) ? g : collect(elem_type(R), g)
  return FreeAssociativeAlgebraIdeal(IdealGens(R, h))
end
```